src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala

```diff
@@ -0,0 +1,135 @@
+package org.allenai.plugins
+
+import sbt.{
+  relativeTo,
+  settingKey,
+  taskKey,
+  AutoPlugin,
+  Def,
+  Keys,
+  PathFinder,
+  Plugins,
+  SettingKey,
+  TaskKey
+}
+import sbt.plugins.JvmPlugin
+
+import java.io.File
+
+/** Plugin for building docker images. */
+object DockerBuildPlugin extends AutoPlugin {
+  val AI2_PRIVATE_REGISTRY = "allenai-docker-private-docker.bintray.io"
+
+  val DEFAULT_BASE_IMAGE = AI2_PRIVATE_REGISTRY + "/java:8"
+
+  /** Requires the JvmPlugin, since this will be building a jar dependency tree. */
+  override def requires: Plugins = JvmPlugin
+
+  object autoImport {
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // The following settings affect both the images and Dockerfiles generated by this plugin. When
+    // you update these settings in a build.sbt file, you'll want to re-generate your Dockerfiles.
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+
+    val dockerfileLocation: SettingKey[File] = settingKey[File](
+      "The location of the Dockerfile to use in building the main project image. Defaults to " +
+        "`srcDirectory.value + \"docker/Dockerfile\", typically \"src/main/docker/Dockerfile\"."
+    )
+
+    // The following three settings control how the generated image is tagged. The image portion of
+    // image tags will be, for the main image:
+    //   ${imageRegistryHost}/${imageNamePrefix}/${imageName}
+    // and for the dependency image will be:
+    //   ${imageRegistryHost}/${imageNamePrefix}/${imageName}-dependency
+    //
+    // See the documentation for details on which tags will be used by `dockerBuild` and
+    // `dockerPush`.
+    val imageRegistryHost: SettingKey[String] = settingKey[String](
+      "The base name of the image you're creating. Defaults to " +
+        "allenai-docker-private-docker.bintray.io ."
+    )
+    val imageNamePrefix: SettingKey[String] = settingKey[String](
+      "The image name prefix (\"repository\", in Docker terms) of the image you're creating. " +
+        "Defaults to organization.value.stripPrefix(\"org.allenai.\") . " +
+        "This is typically the github repository name."
+    )
+    val imageName: SettingKey[String] = settingKey[String](
```

Comment:
Will this be prefixed with something when used?  `imageName` doesn't scream "Docker plugin" to me, similar feedback for other names.  Perhaps they should all be prefixed `docker` like `dockerWorkdir` below?